src/utils/cid/mod.rs

```diff
@@ -42,27 +42,40 @@ pub const BLAKE2B256_SIZE: usize = 32;
 ///
 /// The `Generic` variant is used for CIDs that do not fit into the other variants.
 /// These variants are used for optimizing storage of CIDs in the `FrozenCids` structure.
-#[cfg_attr(test, derive(derive_quickcheck_arbitrary::Arbitrary))]
 #[derive(Clone, Debug, PartialEq, Eq, Hash)]
 pub struct SmallCid(SmallCidInner);
 
 impl SmallCid {
     pub fn cid(&self) -> Cid {
         match &self.0 {
-            SmallCidInner::Generic(cid) => **cid,
+            SmallCidInner::Other(cid) => **cid,
             SmallCidInner::V1DagCborBlake2b(digest) => Cid::new_v1(
                 DAG_CBOR,
                 multihash::Multihash::wrap(Blake2b256.into(), digest)
                     .expect("failed to convert Blake2b digest to Multihash for creation of V1 DAG-CBOR Blake2b CID"),
             ),
         }
     }
+
+    // We always want to represent a CID with the minimal size of `SmallCidInner` if possible, so we can convert the `SmallCid` to its minimal form by checking if the `SmallCidInner` is already in minimal form and converting if necessary.
+    /// [`SmallCidInner::Other`] should not contain a CID which could be represented by more specialized variants.
+    fn canonical_small(cid: Cid) -> SmallCid {
```

Comment:
```suggestion
    fn canonical_small(cid: Cid) -> SmallCidInner {
```

This should be a member of `SmallCidInner`, not `SmallCid`